src/python_routing_v02/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -1155,19 +1155,32 @@ cpdef object compute_network_structured(
                 MC_Reach(segment_objects, array('l',upstream_ids))
                 )
             
+    # replace initial conditions with gage observations, wherever available
+    cdef int gages_size = usgs_positions_list.shape[0]
+    cdef int gage_i, usgs_position_i
+    cdef int gage_maxtimestep = usgs_values.shape[1]
+
+    if gages_size:
+        for gage_i in range(gages_size):
+            usgs_position_i = usgs_positions_list[gage_i]
+            flowveldepth_nd[usgs_position_i, 0, 0] = usgs_values[gage_i, 0]
```

Comment:
Populating initial conditions with gage data (where available). This is basically a straight copy from `mc_reach.compute_network` with slight changes to account for the different indexing scheme used in `mc_reach.compute_network_structured`